### ISSUE TITLE
Cannot install Bonjour on windows 10

### ISSUE THREAD

**Issue by AngeloLazzari** [input-leap/input-leap#2095]
### What happened?

Hy guys, thank you very much for input-leap!! I want to install on my Mac os Sonoma 14.6 as server and windows 10 as client. All is configured ok, the IPs, I tested the cliente can ping and can have a tcp communication with 24800 port, I can start the client, and when i click configur server to set the position of the client, I cannot see any windows instance... 

I don't know how to fix it... can you please help me?  

### Version

v3.0.2

### If applicable, where did you install Input Leap from?

installed from git

### What OSes are you seeing the problem on? (Check all that apply)

macOS

### What OS versions are you using?

Sonoma 14.6

**Comment by slrgt** [comment]
also happens on windows 11 it keeps trying to install it even though its already installed

**Comment by martylamb** [comment]
> also happens on windows 11 it keeps trying to install it even though its already installed

+1

**Comment by rabid-dev** [comment]
+1

**Comment by labroid** [comment]
+1

**Comment by clankill3r** [comment]
+1 windows 11

**Comment by ImOnMars** [comment]
+1
Happens on both my machines (Windows 10 & 11). Anyone know of a workaround? I just installed for the first time

**Comment by hype1type** [comment]
l think https://binaries.symless.com/Bonjour64.msi can't be accessed

**Comment by nbolton** [comment]
> l think https://binaries.symless.com/Bonjour64.msi can't be accessed

Upstream, we [dropped Bonjour](https://github.com/deskflow/deskflow/wiki/Project-Forks#:~:text=Input%20Leap%20(and%20Barrier)%20use%20the%20legacy%20auto%2Ddiscovery%20from%20Synergy%20(Bonjour)%20which%20was%20removed%20from%20Synergy%20due%20to%20issues%20with%20stability.) support _a long time ago_.

**Comment by weiluntong** [comment]
I think a worthwhile note would be that also, Windows 11 Local Security Authority blocks and disables Bonjour.

I don't imagine Apple has any interest in continuing to support Windows (at least where discovery is concerned), as their Windows Store version doesn't even package Bonjour anymore, and the work around to get the latest Bonjour is to *not* go to the windows store, and instead go download the MSI from apple's official site, unzip it with your favorite archiving application, and running the stand-alone Bonjour install. Or just install the iTunes MSI, which will also install Bonjour. Who knows how long this work around will stay though, and again, Bonjour as a whole, even from this workaround, is blocked by Windows 11 LSA.

It might be worthwhile to implement using mDNS features Windows now ships with. Or, hell, might have to go with @nbolton's suggestion in just getting rid of it. There's not a lot of information on whether or not the service discovery portion of Window's zeroconf implementation actually works or if its API is easy to work with.

I guess the tiny push back I can give though is that mDNS makes it quite portable and convenient to bring my laptop and tablet into different environments and just pick back up without having to reconfigure.

**Comment by nbolton** [comment]
> implement using mDNS

I agree. Bonjour is just a wrapper for mDNS anyway, so there's no reason really to use it when you can simply implement it directly. The issue we have is that Qt doesn't (AFAIK) have any mDNS support, so we'd have to use some lib, and C++ is a bit if a wild west for libs. Upstream, we are phasing out all deps except Qt because they're just a right faff to manage.

@sithlord48 ^

**Comment by sithlord48** [comment]
The whole discovery thing never made any sense to me. Its a way over complicated solution for what should simply be the user entering a peer's ip or name.

**Comment by nbolton** [comment]
> should simply be as simple as the user entering a peer's ip or name.

Fair point. Anyone using Input Leap, etc, is likely to be technical enough to use an IP address.

The best thing Input Leap can do is copy upstream and rip Bonjour out altogether.

**Comment by jingjiehaung** [comment]
<img width="883" height="405" alt="Image" src="https://github.com/user-attachments/assets/32584e4d-4287-44d0-9cdc-bf807cdf8411" />
I also encountered the issue of downloading Bonjour, but I tried unchecking the "auto config" option, and then it didn't download Bonjour.
Hope this helps everyone. Thanks
